src/adapter/src/catalog/apply.rs

```diff
@@ -69,207 +59,72 @@ impl CatalogState {
-                    awaiting_name_dependencies
-                        .entry(name)
-                        .or_default()
-                        .push((item, diff));
-                }
-            }
-        }
-
-        // Error on any unsatisfied dependencies.
-        if let Some((missing_dep, mut dependents)) = awaiting_id_dependencies.into_iter().next() {
-            let (
-                mz_catalog::durable::Item {
-                    id,
-                    oid: _,
-                    schema_id,
-                    name,
-                    create_sql: _,
-                    owner_id: _,
-                    privileges: _,
-                },
-                diff,
-            ) = dependents.remove(0);
-            let schema = self.find_non_temp_schema(&schema_id);
-            let name = QualifiedItemName {
-                qualifiers: ItemQualifiers {
-                    database_spec: schema.database().clone(),
-                    schema_spec: schema.id().clone(),
-                },
-                item: name,
-            };
-            let name = self.resolve_full_name(&name, None);
-            let action = if diff == 1 { "deserialize" } else { "remove" };
-
-            return Err(Error::new(ErrorKind::Corruption {
-                detail: format!(
-                    "failed to {} item {} ({}): {}",
-                    action,
-                    id,
-                    name,
-                    plan::PlanError::InvalidId(missing_dep)
-                ),
-            }));
-        }
-
-        if let Some((missing_dep, mut dependents)) = awaiting_name_dependencies.into_iter().next() {
-            let (
-                mz_catalog::durable::Item {
-                    id,
-                    oid: _,
-                    schema_id,
-                    name,
-                    create_sql: _,
-                    owner_id: _,
-                    privileges: _,
-                },
-                diff,
-            ) = dependents.remove(0);
-            let schema = self.find_non_temp_schema(&schema_id);
-            let name = QualifiedItemName {
-                qualifiers: ItemQualifiers {
-                    database_spec: schema.database().clone(),
-                    schema_spec: schema.id().clone(),
-                },
-                item: name,
-            };
-            let name = self.resolve_full_name(&name, None);
-            let action = if diff == 1 { "deserialize" } else { "remove" };
-            return Err(Error::new(ErrorKind::Corruption {
-                detail: format!(
-                    "failed to {} item {} ({}): {}",
-                    action,
-                    id,
-                    name,
-                    Error {
-                        kind: ErrorKind::Sql(SqlCatalogError::UnknownItem(missing_dep))
-                    }
-                ),
-            }));
+            self.apply_update(kind, diff);
```

Comment:
FWIW, soon I will move the sorting of updates from `open.rs` to this function, which I think would make a lot more sense.